File: frontend/lib/features/habits/domain/entities/habit_daily_tracking.dart
import 'package:equatable/equatable.dart';

class HabitDailyTracking extends Equatable {
  final String id;
  final String userId;
  final String habitId;
  final DateTime datetime;
  final double quantityPerSet;
  final int quantityOfSet;
  final String unitId;
  final int weight;
  final String weightUnitId;

  HabitDailyTracking({
    required this.id,
    required this.userId,
    required this.habitId,
    required this.datetime,
    required this.quantityPerSet,
    required this.quantityOfSet,
    required this.unitId,
    required this.weight,
    required this.weightUnitId,
  });

  @override
  List<Object> get props => [
        id,
        userId,
        habitId,
        datetime,
        quantityPerSet,
        quantityOfSet,
        unitId,
        weight,
        weightUnitId,
      ];

  HabitDailyTracking copyWith({
    String? id,
    String? userId,
    String? habitId,
    DateTime? datetime,
    double? quantityPerSet,
    int? quantityOfSet,
    String? unitId,
    int? weight,
    String? weightUnitId,
  }) {
    return HabitDailyTracking(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      habitId: habitId ?? this.habitId,
      datetime: datetime ?? this.datetime,
      quantityPerSet: quantityPerSet ?? this.quantityPerSet,
      quantityOfSet: quantityOfSet ?? this.quantityOfSet,
      unitId: unitId ?? this.unitId,
      weight: weight ?? this.weight,
      weightUnitId: weightUnitId ?? this.weightUnitId,
    );
  }
}

File: backend/src/features/habits/structs/models/habit_daily_tracking.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[allow(non_snake_case)]
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct HabitDailyTracking {
    pub id: Uuid,
    pub user_id: Uuid,
    pub habit_id: Uuid,
    pub datetime: DateTime<Utc>,
    pub created_at: DateTime<Utc>,

    pub quantity_per_set: i32,
    pub quantity_of_set: i32,
    pub unit_id: Uuid,
    pub weight: i32,
    pub weight_unit_id: Uuid,
}

#[allow(non_snake_case)]
#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct HabitDailyTrackingData {
    pub id: Uuid,
    pub user_id: Uuid,
    pub habit_id: Uuid,
    pub datetime: DateTime<Utc>,

    pub quantity_per_set: i32,
    pub quantity_of_set: i32,
    pub unit_id: Uuid,
    pub weight: i32,
    pub weight_unit_id: Uuid,
}

impl HabitDailyTracking {
    pub fn to_habit_daily_tracking_data(&self) -> HabitDailyTrackingData {
        HabitDailyTrackingData {
            id: self.id,
            user_id: self.user_id,
            habit_id: self.habit_id,
            datetime: self.datetime,
            quantity_of_set: self.quantity_of_set,
            quantity_per_set: self.quantity_per_set,
            unit_id: self.unit_id,
            weight: self.weight,
            weight_unit_id: self.weight_unit_id,
        }
    }
}
